src/libpmemobj/palloc.c

```diff
@@ -140,6 +140,16 @@ alloc_prep_block(struct palloc_heap *heap, const struct memory_block *m,
 
 	m->m_ops->write_header(m, extra_field, object_flags);
 
+	/*
+	 * Set allocated memory with pattern, if debug.heap.alloc_pattern CTL
+	 * parameter had been set.
+	 */
+	if (unlikely(heap->alloc_pattern != PALLOC_CTL_DEBUG_NO_PATTERN)) {
```

Comment:
Since `alloc_pattern` is a signed integer, we probably should skip this for all values < 0. AFAIK casting from signed int to char is implementation defined.

Reply:
done